tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -12283,21 +12283,37 @@ void Sema::CheckHLSLFunctionCall(FunctionDecl *FDecl, CallExpr *TheCall,
   }
 }
 
+static uint32_t GetIntConstAttrArg(ASTContext &astContext, const Expr *expr,
+                                   uint32_t defaultVal = 0) {
```

Comment:
```suggestion
static uint32_t GetIntConstAttrArg(ASTContext &astContext, const Expr *expr,
                                   uint32_t defaultVal) {
```

It is really unfortunate to completely duplicate this code, and I don't think the two should be identical. In this version in Sema we should be surfacing an error if there is an expression that isn't a constant expression or spec constant.